webapp/src/app/auth/services/auth.service.ts

```diff
@@ -37,6 +49,25 @@ export class AuthService {
       .pipe(map(res => res.data));
   }
 
+  loginWithProvider(data: UserLoginWithProvider) {
+    return this.http
+      .post<Payload<{ accessToken: string }>>(`${this.endpoint}/auth/token`, {
+        ...data,
+        grantType: 'provider',
+      })
+      .pipe(map(res => res.data));
+  }
+
+  redirectWithProvider({ type, provider }: { type: 'login' | 'signup'; provider: Provider }) {
+    window.open(
+      `${provider.url}?redirect_uri=${provider.redirectUrl}&state=${type}&client_id=${
+        provider.clientId
+      }&scope=email profile openid&access_type=offline&prompt=select_account&response_type=code`,
+      '_blank',
```

Comment:
I think that if you use `_self` in the window.open target, it would then simply some of the message passing by opening the login in the same tab.

Reply:
@anthonynsimon I have a version working with _self without using postMessage but that doesn't open a new window. What did you prefer?